// Copyright 2018 The Paddle Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// =============================================================================

syntax = "proto3";
option optimize_for = LITE_RUNTIME;

package paddlehub.module.checkinfo;

enum FILE_TYPE {
  FILE=0;
  DIR=1;
}

enum REQUIRE_TYPE {
  PYTHON_PACKAGE=0;
  HUB_MODULE=1;
  SYSTEM=2;
  COMMAND=3;
  PY_VERSION=4;
}

message FileInfo {
  string file_name = 1;
  FILE_TYPE type = 2;
  bool is_need = 3;
  string md5 = 4;
  string description = 5;
}

message Requires {
  REQUIRE_TYPE require_type = 1;
  string version = 2;
  bool great_than = 3;
  string description = 4;
}

message CheckInfo {
  string paddle_version = 1;
  string hub_version = 2;
  string module_proto_version = 3;
  repeated FileInfo file_infos = 4;
  repeated Requires requires = 5;
};
